pageserver/src/http/routes.rs

```diff
@@ -1912,12 +1912,15 @@ async fn timeline_collect_keyspace(
         let ctx = RequestContext::new(TaskKind::MgmtRequest, DownloadBehavior::Download);
         let timeline = active_timeline_of_active_tenant(&state.tenant_manager, tenant_shard_id, timeline_id).await?;
         let at_lsn = at_lsn.unwrap_or_else(|| timeline.get_last_record_lsn());
-        let keys = timeline
+        let (dense_ks, _) = timeline
             .collect_keyspace(at_lsn, &ctx)
             .await
             .map_err(|e| ApiError::InternalServerError(e.into()))?;
 
-        let res = pageserver_api::models::partitioning::Partitioning { keys, at_lsn };
+        // This API is currently used by pagebench. Pagebench will iterate all keys within the keyspace.
+        // Therefore, we only return dense keyspace for now.
+
+        let res = pageserver_api::models::partitioning::Partitioning { keys: dense_ks, at_lsn };
```

Comment:
To avoid future confusion, it would be nice to include sparse ranges here as well, perhaps just as an extra field on `Partitioning`.  Otherwise I can imagine some future developer looking at the output of this API and wondering where our aux file keys are.

Reply:
fixed in https://github.com/neondatabase/neon/pull/7503/commits/2d87183bd048e7a0b60ffb57dfef23f6f1714d6f